impl/src/main/java/io/kroxylicious/testing/kafka/common/BrokerCluster.java

```diff
@@ -60,10 +65,9 @@
  *     public void testKafkaClusterParameter(
  *             @BrokerCluster(numBrokers = 3)
  *             @KRaftCluster(numControllers = 3)
- *             @SaslPlainAuth({
- *                 @UserPassword(user="alice", password="foo"),
- *                 @UserPassword(user="bob", password="bar")
- *             })
+ *             @User(user="alice", password="foo")
+ *             @User(user="bob", password="bar")
+ *             @SaslMechanism("SCRAM-SHA-256")
```

Comment:
I agree the current APIs are probably a bit off.   I think clusters should probably have `@Listener` and those should have a optional SASL mechanism property and a list of users.    Listeners should probably have a @Name which can then be referred to by an annotation on the injected the Kafka Client.  

```java
@Test
void myTest(@Listener(name="foo", saslMechanism="SCRAM-SHA-512 @User("alice", "secret)) KafkaCluster cluster, @ListenerRef(name="foo", userRef="alice") admin) {
...
}
```

Unfortunately, I don't think we have the time to spend right now on this.  I think what we have is serving us well enough.  Having the ability to test other SASL mechanisms will be a enabler for the Kroxylicious project.

I do hope we get time to revisit this again.  I'd love to see the Junit5 ext get traction outside Kroxylicious.  I think it has got real value for anyone integration testing with Kafka.

Reply:
actually the idea of making `@User` a child of `@Listener` is duff.  SCRAM users are scoped globally to the Cluster.  They are not exclusive to the listener.   (For PLAIN, you can have a `bob` with a different password at each listener.  But that's just weird.. it is still the same bob..  maybe we don't need to model that).

making User a child of SaslMechanism would be a better.

@SaslMechanism(mechanism="PLAIN", @User("alice", "secret")...)

that would also leave the door open for a `@Listener` annotation   `@Listener(mechanism="PLAIN" name="foo")`.  I'm not suggesting adding a @Listener now.

Reply:
> making User a child of SaslMechanism

@SamBarker what would you think to this idea?

Reply:
I like the idea 

```java
@BrokerCluster
@SaslMechanism(mechanism = "SCRAM-SHA-512",
               principals = { @Principal(identity = "alice", secret = "foo"), @Principal(identity = "bob", secret = "bar") })
KafkaCluster cluster
```

Feels like a good (if verbose) model. `SaslMechanism` possibly also needs a name so that we could attach them to listeners if we went down that route (but we should probably apply YAGNI and add it later...)

Reply:
Thanks, I'll give that a go